src/test/java/seedu/letsgethired/logic/commands/UndoCommandTest.java

```diff
@@ -0,0 +1,89 @@
+package seedu.letsgethired.logic.commands;
+
+import static seedu.letsgethired.logic.commands.CommandTestUtil.VALID_COMPANY_BYTEDANCE;
+import static seedu.letsgethired.logic.commands.CommandTestUtil.VALID_ROLE_BACK_END;
+import static seedu.letsgethired.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.letsgethired.testutil.TypicalInternApplications.getTypicalInternTracker;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.letsgethired.model.Model;
+import seedu.letsgethired.model.ModelManager;
+import seedu.letsgethired.model.UserPrefs;
+import seedu.letsgethired.model.application.InternApplication;
+import seedu.letsgethired.testutil.InternApplicationBuilder;
+
+
+public class UndoCommandTest {
+    private Model model = new ModelManager(getTypicalInternTracker(), new UserPrefs());
+
+    @Test
+    public void execute_afterAddCommand_success() {
+        InternApplication validInternApplication = new InternApplicationBuilder().build();
+        UndoCommand undoCommand = new UndoCommand();
+
+        String expectedMessage = UndoCommand.MESSAGE_SUCCESS_UNDONE;
+        ModelManager expectedModel = new ModelManager(model.getInternTracker(), new UserPrefs());
+
+        model.addInternApplication(validInternApplication);
+        assertCommandSuccess(undoCommand, model, expectedMessage, expectedModel);
```

Comment:
Perhaps it would be better to actually execute the command being undone for the test cases?

I think adding a helper method in `CommandTestUtil` to test undos can also be useful!
```java
    public static void assertUndoSuccess(Command command, UndoCommand undoCommand, Model actualModel,
                                         String expectedMessage, Model expectedModel) {
        try {
            command.execute(actualModel);
            assertNotEquals(expectedModel, actualModel);
            undoCommand.execute(actualModel);
            assertCommandSuccess(undoCommand, actualModel, expectedMessage, expectedModel);
        } catch (CommandException ce) {
            throw new AssertionError("Execution of command should not fail.", ce);
        }
    }
```

Feel free to ignore these suggestions if you prefer the current approach :D